### PR TITLE
Draw distortion_1D figure

### PR DESCRIPTION
**Pull request by surrender2002**

### CODE REVIEW

File: deeplens/geolens.py [hunk 1]
```diff
@@ -1429,9 +1429,738 @@ def distortion(self, depth=DEPTH, grid_size=64):
...
+                    best_position, best_distance
+                )
+            
+            return best_position, fixed_direction
+            
+        finally:
+            # Restore original precision
+            torch.set_default_dtype(orig_dtype)
+
+    def _calc_chief_ray_infinite_dls(self, fov, wvln, max_iterations, tolerance, plane,
```
Comment (singer-yang) [inline deeplens/geolens.py:1588]: This function is too long

File: deeplens/optics/surfaces.py [hunk 2]
```diff
@@ -191,12 +191,12 @@ def refract(self, ray, eta):
         new_d = sr * n + eta * (ray.d - cosi.unsqueeze(-1) * n)
 
         # Update ray direction
-        new_d[~valid] = ray.d[~valid]
+        new_d[~valid] = ray.d[~valid].to(new_d.dtype)
```
Comment (singer-yang) [inline deeplens/optics/surfaces.py:194]: Usually we do not want to explicitely change the tensor dtype. If there is a dtype mismatch, it should be fixed somewhere else, for example, when creating the tensor, specify the dtype.

**Review by singer-yang — CHANGES_REQUESTED** [review]

File: deeplens/geolens.py (hunk 1 above)
Comment (singer-yang) [reply deeplens/geolens.py]: The functions have been rewrited by Xinge

File: deeplens/geolens.py [hunk 3]
```diff
@@ -1741,6 +1793,69 @@ def chief_ray(self):
...
+        # Scale factor
+        pupilz, pupilx = self.calc_entrance_pupil()
+        scale = 0.45
+        delta = scale * pupilx * 2
+
+        # Sample parallel rays from object space
+        y_distance = math.tan(hfov) * (10.0 + pupilz)
+        min_y = (-pupilx - y_distance) + delta
+        max_y = (pupilx - y_distance) - delta
+        o1 = torch.zeros([SPP_CALC, 3])
```
Comment (LinYark) [inline deeplens/geolens.py:1836]: -10 should be proposed as a constant variable

Comment (singer-yang) [reply deeplens/geolens.py:1836]: Better construct the ray, then propagate to object distance (for example, d = -10.0), following the geolens.sample_parallel() function.

**Review by LinYark — CHANGES_REQUESTED** [review]
@Xinge, consider whether to add a function for calculating the entrance pupil with a near axis？

**Comment by singer-yang** [comment]
I have some recent updates for entrance pupil calculation. Lets merge first, then update the paraxial entrance pupil calculation.
